orchestra/communication/staffing.py

```diff
@@ -188,14 +275,26 @@ def send_request_inquiries(staffbot, request, worker_batch_size):
     worker_certifications = (
         WorkerCertification
         .objects
-        .exclude(worker__id__in=workers_with_inquiries)
         .filter(role=required_role,
                 task_class=WorkerCertification.TaskClass.REAL,
                 certification__in=(request.task
                                    .step.required_certifications.all()))
         .order_by('-staffing_priority', '?'))
-    _send_request_inquiries(staffbot, request, worker_batch_size,
-                            worker_certifications)
+    available_worker_certifications = (
+        worker_certifications
+        .filter(worker__availabilities__week=first_day_of_the_week()))
+    uninquired_worker_certifications = (
+        worker_certifications
+        .exclude(worker__id__in=workers_with_inquiries))
+    successfully_staffed = _attempt_to_automatically_staff(
+        staffbot, request, available_worker_certifications)
+    sending_inquiries = StaffBotRequest.Status.SENDING_INQUIRIES.value
+    if ((not successfully_staffed)
+            and (request.status == sending_inquiries)):
```

Comment:
why do we need the request.status check? Can successfully_staffed be false, but somehow we altered the request's status during autostaffing anyways?

Reply:
Add a comment:

We consider StaffBotRequests that are done sending inquiries when auto-staffing, since it's possible for a worker to have auto-staffing availability for a request that has already been sent to them (e.g., the day after receiving a request when new availability kicks in). But we only want to send new request inquiries for requests that aren't `DONE_SENDING_INQUIRIES`.